tests/java/org/python/core/ReflectedArgsTest.java

```diff
@@ -1,53 +1,167 @@
+        assertTrue(FOO_SIGNATURE.matches(null, pyArgs, Py.NoKeywords, callData));
+        fooCheck(callData.args, new Object[] {3, 4, strlist});
     }
 
-    public void testVarargsBoxedWithTooFewArguments() {
+    /**
+     * Calling {@code bar(int a, int[] b, int... c)} as Python {@code bar(1, [2, 3])} calls Java
+     * {@code bar(1, new int[]{2, 3})}.
+     */
+    public void testPrimitiveVarargsBoxedWithTooFewArguments() {
         // calling bar(int a, int[] b, int... c) as bar(1, [2, 3])
-        ReflectedArgs args = new ReflectedArgs(null, new Class<?>[] {Integer.class, Integer[].class, Integer[].class}, Demo.class, true, true);
-        PyObject[] ints = new PyObject[] {Py.newInteger(3), Py.newInteger(4)};
-        assertTrue(args.matches(null, new PyObject[] {Py.newInteger(1), new PyList(ints)}, new String[0], new ReflectedCallData()));
+        PyObject[] ints = {Py.newInteger(2), Py.newInteger(3)};
+        PyObject[] pyArgs = {Py.newInteger(1), new PyList(ints)};
+        ReflectedCallData callData = new ReflectedCallData();
+        assertTrue(BAR_SIGNATURE.matches(null, pyArgs, Py.NoKeywords, callData));
+        barCheck(callData.args, 1, new int[] {2, 3});
     }
 
+    /**
+     * Calling {@code baz(Integer a, Integer[] b, Integer... c)} as Python {@code baz(1, [2, 3])}
+     * calls Java {@code baz(1, new Integer[]{2, 3})}.
+     */
+    public void testVarargsBoxedWithTooFewArguments() {
+        // calling baz(Integer a, Integer[] b, Integer... c) as bar(1, [2, 3])
+        PyObject[] ints = {Py.newInteger(2), Py.newInteger(3)};
+        PyObject[] pyArgs = {Py.newInteger(1), new PyList(ints)};
+        ReflectedCallData callData = new ReflectedCallData();
+        assertTrue(BAZ_SIGNATURE.matches(null, pyArgs, Py.NoKeywords, callData));
+        bazCheck(callData.args, 1, new Integer[] {2, 3});
+    }
+
+    /**
+     * Calling {@code foo(Object... args)} as Python {@code foo([1,2,3])} calls Java {@code foo(new
+     * Object[]{1,2,3})}.
+     */
     public void testVarargsNotBoxedWithCorrectArgs() {
         // calling foo(Object... args) as foo([1,2,3])
-        ReflectedArgs args = new ReflectedArgs(null, new Class<?>[] {Object[].class}, Demo.class, true, true);
-        PyObject[] varargs = new PyObject[] {Py.newString("foo"), Py.newString("bar") };
-        PyObject[] argList = new PyObject[] {new PyList(varargs)};
-        assertTrue(args.matches(null, argList, new String[0], new ReflectedCallData()));
+        PyObject[] ints = {Py.newInteger(1), Py.newInteger(2), Py.newInteger(3)};
+        PyObject[] pyArgs = {new PyList(ints)};
+        ReflectedCallData callData = new ReflectedCallData();
+        assertTrue(FOO_SIGNATURE.matches(null, pyArgs, Py.NoKeywords, callData));
+        fooCheck(callData.args, 1, 2, 3);
     }
 
+    /**
+     * Calling {@code foo(Object... args)} as Python {@code foo("foo", "bar")} calls Java
+     * {@code foo("foo", "bar")}.
```

Comment:
should this be `calls Java {@code foo(new String[] {"foo", "bar"})}`?

Reply:
I think I wrote `{@code foo(new Object[] {"foo", "bar"})}` first, but as that is quite hard to read and `foo` is a varargs in Java, I think what I have is ok.